test/e2e/auth/selfsubjectreviews.go

```diff
@@ -0,0 +1,119 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo/v2"
+	authenticationv1alpha1 "k8s.io/api/authentication/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("SelfSubjectReview [Feature:APISelfSubjectReview]", func() {
+	f := framework.NewDefaultFramework("selfsubjectreviews")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	/*
+		Release: v1.24
+		Testname: SelfSubjectReview API
+		Description:
+		The authentication.k8s.io API group MUST exist in the /apis discovery document.
+		The authentication.k8s.io/v1alpha1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
+		The selfsubjectreviews resource MUST exist in the /apis/authentication.k8s.io/v1alpha1 discovery document.
+		The selfsubjectreviews resource must support create.
+	*/
+	ginkgo.It("should support SelfSubjectReview API operations", func() {
+		// Setup
+		ssarAPIVersion := "v1alpha1"
+
+		// Discovery
+		ginkgo.By("getting /apis")
+		{
+			discoveryGroups, err := f.ClientSet.Discovery().ServerGroups()
+			framework.ExpectNoError(err)
+			found := false
+			for _, group := range discoveryGroups.Groups {
+				if group.Name == authenticationv1alpha1.GroupName {
+					for _, version := range group.Versions {
+						if version.Version == ssarAPIVersion {
+							found = true
+							break
+						}
+					}
+				}
+			}
+			if !found {
+				framework.Failf("expected SelfSubjectReview API group/version, got %#v", discoveryGroups.Groups)
+			}
+		}
+
+		ginkgo.By("getting /apis/authentication.k8s.io")
+		{
+			group := &metav1.APIGroup{}
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath("/apis/authentication.k8s.io").Do(context.TODO()).Into(group)
+			framework.ExpectNoError(err)
+			found := false
+			for _, version := range group.Versions {
+				if version.Version == ssarAPIVersion {
+					found = true
+					break
+				}
+			}
+			if !found {
+				framework.Failf("expected SelfSubjectReview API version, got %#v", group.Versions)
+			}
+		}
+
+		ginkgo.By("getting /apis/authentication.k8s.io/" + ssarAPIVersion)
+		{
+			resources, err := f.ClientSet.Discovery().ServerResourcesForGroupVersion(authenticationv1alpha1.SchemeGroupVersion.String())
+			framework.ExpectNoError(err)
+			found := false
+			for _, resource := range resources.APIResources {
+				switch resource.Name {
+				case "selfsubjectreviews":
+					found = true
+				}
+			}
+			if !found {
+				framework.Failf("expected selfsubjectreviews, got %#v", resources.APIResources)
```

Comment:
I expected to see a ginkgo.Skip when the resource was missing instead of a fail.  for some reason, this test isn't running even in alpha tests.

Reply:
Skip is added (I also tested it manually).